geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedFeatureSourceView.scala

```diff
@@ -46,11 +48,18 @@ class MergedFeatureSourceView(
   override def getSchema: SimpleFeatureType = sft
 
   override def getCount(query: Query): Int = {
-    // if one of our sources can't get a count (i.e. is negative), give up and return -1
-    val total = sources.foldLeft(0) { case (sum, (source, filter)) =>
-      lazy val count = source.getCount(mergeFilter(query, filter))
-      if (sum < 0 || count < 0) { -1 } else { sum + count }
-    }
+    val total =
+      if (parallel) {
+        val counts = sources.par.map { case (source, f) => source.getCount(mergeFilter(query, f)) }
+        counts.foldLeft(0)((sum, count) => if (sum < 0 || count < 0) { -1 } else { sum + count })
```

Comment:
The only unfortunate thing about this parallel computation is that all the counts will be computed even if one of the sources returns a -1, right?

Reply:
yeah... we could map them to futures and try to cancel them when we see a -1, but i think that would likely not work great given all the effort we had to put in to get query timeouts working.